src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationTask.kt

```diff
@@ -198,11 +199,26 @@ class ShardReplicationTask(id: Long, type: String, action: String, description:
         val followerIndexService = indicesService.indexServiceSafe(followerShardId.index)
         val indexShard = followerIndexService.getShard(followerShardId.id)
 
+
         try {
             //Retention leases preserve the operations including and starting from the retainingSequenceNumber we specify when we take the lease .
             //hence renew retention lease with lastSyncedGlobalCheckpoint + 1
             retentionLeaseHelper.renewRetentionLease(leaderShardId, indexShard.lastSyncedGlobalCheckpoint + 1, followerShardId)
-        } catch (ex: Exception) {
+        } catch (ex: RetentionLeaseNotFoundException) {
+           //have to simplify this
+            try {
+                var oldRetentionLeaseHelper = RemoteClusterRetentionLeaseHelper( clusterService.clusterName.value(), remoteClient)
+                if (oldRetentionLeaseHelper.verifyRetentionLeaseExist(leaderShardId, followerShardId)) {
+                    // Retention lease is present
+                    retentionLeaseHelper.addRetentionLease(leaderShardId,indexShard.lastSyncedGlobalCheckpoint + 1,followerShardId, RemoteClusterRepository.REMOTE_CLUSTER_REPO_REQ_TIMEOUT_IN_MILLI_SEC)
+                } else {
+                    throw Exception("Old retention ID not found")
```

Comment:
Can you explain why these changes are required?

Reply:
When replication is not explicitly paused and version upgrades happen on follower, this flow helps add new retention lease.